R/format_GLA.R

```diff
@@ -0,0 +1,638 @@
+
+    rr_data <- rr_data %>%
+      dplyr::filter(.data$Species %in% species_filter & !(is.na(.data$Species)))
+
+  }
+
+  ## Filter to keep only desired Populations if specified
+  if(!is.null(pop_filter)){
+
+    nest_data <- nest_data %>%
+      dplyr::filter(.data$PopID %in% pop_filter & !(is.na(.data$PopID)))
+
+    rr_data <- rr_data %>%
+      dplyr::filter(.data$PopID %in% pop_filter & !(is.na(.data$PopID)))
+
+  }
+
+
+  #### BROOD DATA
+  message("Compiling brood information...")
+  Brood_data <- create_brood_GLA(data)
+
+  #### CAPTURE DATA
+  message("Compiling capture information...")
+  Capture_data <- create_capture_GLA(data)
+
+  #### INDIVIDUAL DATA
+  message("Compiling individual information...")
+  Individual_data <- create_individual_GLA(Capture_data)
+
+  #### LOCATION DATA
+  message("Compiling location information...")
+  Location_data <- create_location_GLA(data)
+
+  time <- difftime(Sys.time(), start_time, units = "sec")
+
+  message(paste0("All tables generated in ", round(time, 2), " seconds"))
+
+  #### EXPORT DATA
+
+  if(output_type == "csv"){
+
+    message("Saving .csv files...")
+
+    utils::write.csv(x = Brood_data, file = paste0(path, "\\Brood_data_GLA.csv"), row.names = F)
+
+    utils::write.csv(x = Capture_data, file = paste0(path, "\\Capture_data_GLA.csv"), row.names = F)
+
+    utils::write.csv(x = Individual_data, file = paste0(path, "\\Individual_data_GLA.csv"), row.names = F)
+
+    utils::write.csv(x = Location_data, file = paste0(path, "\\Location_data_GLA.csv"), row.names = F)
+
+    invisible(NULL)
+
+  }
+
+  if(output_type == "R"){
+
+    message("Returning R objects...")
+
+    return(list(Brood_data = Brood_data,
+                Capture_data = Capture_data,
+                Individual_data = Individual_data,
+                Location_data = Location_data))
+
+  }
+
+}
+
+#### --------------------------------------------------------------------------~
+#### FUNCTIONS
+#### --------------------------------------------------------------------------~
+
+
+#' Create brood data table for great tits and blue tits in Glasgow, Scotland.
+#'
+#' Create brood data table in standard format for Glasgow, Scotland.
+#'
+#' @param data Data frame of modified primary data from Glasgow, Scotland.
+#'
+#' @return A data frame.
+
+create_brood_GLA <- function(nest_data, rr_data) {
+
+  ## Create brood data template
+  load(file = "data/Brood_data_template.rda")
+
+  ## Get brood data from ringing records
+  rr_data_brood_sum <- rr_data %>%
+    dplyr::filter(is.na(.data$LocationID) == F) %>%
+
+    ## Determine whether a chick or adult - done for each observation since there is no unique identifier for individuals
+    dplyr::mutate(RingAge = ifelse(.data$Age_observed == 1, "chick", "adult")) %>%
+
+    ## Keeping chicks
+    dplyr::filter(RingAge == "chick") %>%
+
+    ## Summarize brood information for each nest
+    dplyr::group_by(.data$BreedingSeason, .data$PopID, .data$LocationID) %>%
```

Comment:
I agree though I'll need to think about how to do this since it doesn't seem straightforward to join lay date to the ringing data. Something like a rolling join with capture date and lay date could work, but might be messy.

Reply:
Do we have a solution for this yet? Could consider using `fuzzyjoin`.